server/index.js

```diff
@@ -193,6 +193,36 @@ app.delete('/api/deletecard/:cardId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.delete('/api/deletedeck/:deckId', (req, res, next) => {
+  const deckId = Number(req.params.deckId);
+  if (!deckId) {
+    throw new ClientError(400, 'deckId must be a positive integer');
+  }
+
+  const sql = `
+  WITH deletedCards as (
+    delete from "flashcards"
+      where "deckId" = $1
+        and "userId" = $2
+    )
+    delete from "decks"
+      where "deckId" = $1
+  `;
+
+  const params = [deckId, 1];
+
+  db.query(sql, params)
+    .then(result => {
+      const deleted = result.rows;
+      if (!deleted) {
```

Comment:
We'll need to check if the array is empty, otherwise, this condition will never be true, as even empty arrays are truthy.

Reply:
Gotcha, good to know. I ran into the issue where I was throwing the error even if it deleted something because I had destructured the first element from the rows property of the result object so I had taken it a step back but didn't take into consideration the empty array.

      if (!deleted || deleted === []) {